py/kubeflow/testing/cleanup_ci.py

```diff
@@ -316,39 +313,30 @@ def cleanup_instance_groups(args):
         break
       for s in results["items"]:
         name = s["name"]
-        age = getAge(s["creationTimestamp"])
-
-        infra_type = name_to_infra_type(name)
-
-        if not infra_type:
-          logging.info("Skipping intance group %s; it does not match any "
-                       "infra type.", name)
+        size = s["size"]
+        if size > 0:
+          logging.info("Skipping instance group %s because it is in use by %d "
+                       "instances.", name, size)
+          in_use.append(name)
           continue
 
-        logging.info("Instance group %s categorized as %s", name, infra_type)
-
-        if age > MAX_LIFETIME[infra_type]:
```

Comment:
Why are you removing the max lifetime criterion? What if a GKE cluster orphans an instance group of fixed size how would that end up being GC'd?

Reply:
move the policy back.  to the question, I think as long as an instance group has dependencies running, we cannot delete it.